### ISSUE TITLE
Scaling issue

### ISSUE THREAD

**Issue by haesleinhuepf** [haesleinhuepf/napari-plot-profile#8]
Reported by @psobolewskiPhD  in #3:

1) Sometimes the plots are not what I expect? See example below:
![image](https://user-images.githubusercontent.com/76622105/130432469-2cf9f760-2bc8-4323-8331-8cfb984c52cc.png)
I think I have the line draw fairly symmetrical and mousing over points returns values that are in agreement with the matplotlib graph my widget generates. But your plugin graph is odd, very asymmetrical and just one peak. Is it not actually doing line profiles but something different? Is it due to the angle, so pixels are not neighboring correctly?
For reference, I'm using `measure.profile_line` from skimage for my plot—again based on napari/examples (my code in gist at bottom, with some comments)

Here's a link to my hacked together widget, which should run:
(FYI The examples above were with a different approach where I get the shape layer and slice.)
https://gist.github.com/psobolewskiPhD/53eec7440eda501d9edf08a99659f6b0
Ironically, this example doesn't work with your plugin. Traceback:
<details>
Traceback (most recent call last):
  File "/Users/piotrsobolewski/Dev/miniforge3/envs/TF-StarDist-napari/lib/python3.9/site-packages/napari_plot_profile/_dock_widget.py", line 101, in _on_selection
    self.redraw(force_redraw=True)
  File "/Users/piotrsobolewski/Dev/miniforge3/envs/TF-StarDist-napari/lib/python3.9/site-packages/napari_plot_profile/_dock_widget.py", line 158, in redraw
    my_profile = profile(layer, line, num_points=num_bins)
  File "/Users/piotrsobolewski/Dev/miniforge3/envs/TF-StarDist-napari/lib/python3.9/site-packages/napari_plot_profile/_dock_widget.py", line 219, in profile
    start = line[min(current_line, len(line) - 1)] / layer.scale
ValueError: operands could not be broadcast together with shapes (2,) (3,) 
</details>
This blocks all interaction with the shapes layer and breaks the plugin. Is there a way to *switch off* a plugin? the [x] just closes it, but it's still listed in the Window menu. 

I think because I take the route of making a 2D generic shapes layer that appears on all slices, since for a "tool" it was more intuitive to me to see a line profile for the same physical line in z or t. It helped get around the issue of having to re-draw a line on a different slice if I didn't delete the previous one. I still want advancing slices to trigger the replotting too...Edit: got it working! updated gist.
Just passing this to napari triggers the traceback:

```
line = np.array([[11, 13], [111, 113]])
line_prof_layer = viewer.add_shapes(
    line, shape_type="line", edge_color="red", name="Line Profile"
)
```

**Comment by haesleinhuepf** [comment]
Hi @psobolewskiPhD,

I think the issue might be that you are providing a two-dimensional line but the data-set is three-dimensional. I'm not sure how the shape-layer deals with adding two-dimensional points in case the dataset has more dimensions. Maybe @sofroniewn can comment.

Anyway, I'm having issues reproducing your exact issue.

When running the code you mentioned, I see this line, which doesn't fit to any of the plots in your screenshot.
![image](https://user-images.githubusercontent.com/12660498/130453060-a97a5bdd-cad2-46ea-999a-ccb66972524f.png)

When drawing a line in this dataset by hand, the profile plot looks reasonable.
![image](https://user-images.githubusercontent.com/12660498/130453382-d276961b-6e3c-4168-a454-cf2b323c5789.png)

That's your code I used. If you could refine it so that the issue becomes clearer to me what's going on, that would be great. You know, a [minimal-working-example](https://en.wikipedia.org/wiki/Minimal_working_example) ;-)
```
import matplotlib.pyplot as plt
from matplotlib.backends.backend_qt5agg import FigureCanvas
from skimage import measure
from skimage import io
import numpy as np
import os
import napari

#%%
# # create image
# x = np.linspace(0, 5, 256)
# y = np.linspace(0, 5, 256)[:, np.newaxis]
# img = np.sin(x) ** 10 + np.cos(10 + y * x) * np.cos(x)
#%%
# add it to the viewer
imgName = os.path.expanduser("Bars-G10-P30.tif")
img = io.imread(imgName)
viewer = napari.view_image(img, name="Image", colormap="viridis")
line = np.array([[11, 13], [111, 113]])
line_prof_layer = viewer.add_shapes(
    line, shape_type="line", edge_color="red", name="Line Profile"
)
line_prof_layer.mode = "select"


#%%
def line_profile(image, line_prof_layer):
    slice = viewer.layers["Image"]._data_raw
    if line_prof_layer.data:
        if line_prof_layer.shape_type[-1] == "line":
            linescan = measure.profile_line(
                slice,
                line_prof_layer.data[-1][0],
                line_prof_layer.data[-1][1],
                mode="reflect",
            )
        else:
            print("no line")
    else:
        print("No data")
    return linescan


#%%
# create mpl figure with subplots
plt.rcParams.update({"font.size": 11})
mpl_fig = plt.figure()
ax = mpl_fig.add_subplot(1, 1, 1)
linescan = line_profile(img, line_prof_layer)
(line,) = ax.plot(linescan)
line.figure.canvas.draw()

#%%
# add the figure to the viewer as a FigureCanvas widget
viewer.window.add_dock_widget(FigureCanvas(mpl_fig), area="bottom")

# connect a callback that updates the line plot when
@line_prof_layer.mouse_drag_callbacks.append
def profile_lines_drag(shapes_layer, event):
    linescan = line_profile(img, shapes_layer)
    ax.clear()
    (line,) = ax.plot(linescan)
    line.figure.canvas.draw()
    yield
    while event.type == "mouse_move":
        linescan = line_profile(img, shapes_layer)
        ax.clear()
        (line,) = ax.plot(linescan)
        line.figure.canvas.draw()
        # the yield statement allows the mouse UI to keep working while
        # this loop is executed repeatedly
        yield
        
#%%
# try to connect to slider
@viewer.dims.events.current_step.connect
def profile_lines_slice(event):
    linescan = line_profile(img, line_prof_layer)
    ax.clear()
    (line,) = ax.plot(linescan)
    line.figure.canvas.draw()
```

**Comment by psobolewskiPhD** [comment]
Gah, I'm sorry I just figured out the issue!
It was again the fact that I went back to a slice, so moving the line to the diagonal didn't do anything. I didn't realize the plot didn't update. But launching my widget with that slice open to check grabbed the correct profile. But this is just a coincidence of the order of the operations. And essentially connected to https://github.com/haesleinhuepf/napari-plot-profile/issues/7
Sorry!

The gist that generates the red line is unrelated to the above, I'm sorry I didn't make that more clear. i was just wanting to show you what "makes sense" for me.
Likewise, the traceback error with my code is unrelated to this issue, it's just something that the plugin doesn't handle—a 2D shapes layer. Since napari examples generate such layers, it may be something to catch.
